File: algorithms/insertion-sort/c/insertion-sort.c
#include "insertion-sort.h"

void insertion_sort(int * arr, int length) {
    for (int i = 1; i < length; i++) {
        int j = i - 1;
        while (j >= 0 && arr[j] > arr[j + 1]) {
            // swap values
            int temp = arr[j];
            arr[j] = arr[j + 1];
            arr[j + 1] = temp;
            j--;
        }
    }
}

void print_array(int * arr, int length) {
    for (int i = 0; i < length; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

File: algorithms/insertion-sort/javascript/insertion-sort.mjs
export function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let j = i - 1;
        while (j >= 0 && arr[j] < arr[j + 1]) {
            const temp = arr[j + 1];
            arr[j + 1] = arr[j];
            arr[j] = temp;
            j--;
        }
    }
    return arr;
}

export function insertionSort2(arr) {
    for (let i = 1; i < arr.length; i++) {
        const key = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
    return arr;
}
